src/Api/TMDB/tmdbService.tsx

```diff
@@ -0,0 +1,91 @@
+export const fetchMovieGenres = async () => {
+    const options = {
+        method: "GET"
+    }
+    const url = `${import.meta.env.VITE_TMDB_URL}genre/movie/list?api_key=${import.meta.env.VITE_TMDB_APIKEY}`;
+    const response = await fetch(url, options);
+    const responseData = await response.json();
+    return { status: response.status, data: responseData };
+}
+
+export const fetchMovies = async (pageNumber: number) => {
+    const options = {
+        method: "GET"
+    }
+    const url = `${import.meta.env.VITE_TMDB_URL}movie/top_rated?api_key=${import.meta.env.VITE_TMDB_APIKEY}&page=${pageNumber}`;
+    const response = await fetch(url, options);
+    const responseData = await response.json();
+    return { status: response.status, data: responseData };
+}
+
+export const fetchMovieDetails = async (movieId: number) => {
+    const options = {
+        method: "GET"
+    }
+    const url = `${import.meta.env.VITE_TMDB_URL}movie/${movieId}?api_key=${import.meta.env.VITE_TMDB_APIKEY}`;
+    const response = await fetch(url, options);
+    const data = await response.json();
+    let trailerVideo = await fetchMovieTrailer(movieId);
+    if (trailerVideo !== undefined && trailerVideo !== null) {
+        data.trailer = trailerVideo
+    }
+    let CreditsData = await fetchMovieCast(movieId)
+    if (CreditsData?.directorData.length > 0) {
+        data.directors = CreditsData?.directorData
+    }
+    if (CreditsData?.writerData.length > 0) {
+        data.writers = CreditsData?.writerData
+    }
+    if (CreditsData?.castData.length > 0) {
+        data.cast = CreditsData?.castData
+    }
+    return { status: response.status, data };
+}
```

Comment:
in case of this function, once you create a reusable function for data fetching, get the raw data, then pass through another function that does the transformations you have done.

Reply:
I used the "fetchGetRequest" function to get the data but did all the remaining transformations in the same function